lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +294,130 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, **options)
+    if options[:retention].present? || options[:retention_keep_table] == false
```

Comment:
So you're suggesting that we have keyword args, but exclude `retention` and `retention_keep_table` such that it results in an argument error when called with those options? I think the existing error is _much_ more helpful and consistent with other methods in this library

Reply:
That's a good argument, and I'd thought on that, OTOH that's a _runtime_ error, while the kwargs error would be preventing the code from being valid at all. Though...it's Ruby, so...maybe no editor is going to check that anyway.

Either way I think we should make the args consistent.

Reply:
From a functionality perspective, the only thing we gain by having kwargs is that the argument validation happens before the check for `retention` and `retention_keep_table`. Without kwargs the argument validation happens in the unsafe version of the method and I don't think that's a big deal. I'm just hesitant to do this because delegating to another method when kwargs are present becomes very verbose (and ugly IMO):

```ruby
def foo(a, b:, c:, d:, e:)
  # do some validation then delegate to bar
  bar(a, b: b, c: c, d: d, e: e)
end
```

I know that kwargs provide better documentation for the method signature, but if anyone who is vaguely familiar with ruby is looking at this code, I think it's pretty clear that this method is just delegating and they should reference the unsafe version of the method to see the actual method signature